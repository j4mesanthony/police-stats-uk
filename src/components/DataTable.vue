<script setup lang="ts">
import { useWindowSize } from '../composables/useWindowSize';

const emits = defineEmits(['click']);

const props = defineProps({
    data: {
        type: Object,
        required: true
    },

    key: {
        type: String,
        default: 'id'
    },

    itemSize: {
        type: [Number, String],
        default: 1
    },

    isRowsClickable: {
        type: Boolean,
        default: false
    },
});

const { isMobile } = useWindowSize();
</script>

<template>
    <RecycleScroller
        :key="data.length"
        :items="props.data"
        :item-size="props.itemSize"
        :key-field="props.key"
        v-slot="{ item }">
        
        <div :class="{ 'hover:text-slate-50 hover:border-slate-50 origin-bottom': !isMobile, 'hover:scale-[1.02]' : !isMobile && props.isRowsClickable }" 
              class="p-4 pl-3 pr-3 text-sm text-left transition-all duration-200 border-b cursor-pointer dark:border-slate-700 dark:text-slate-400"
             @click="emits('click', item)">
            {{ item.name }}
        </div>
    </RecycleScroller>
</template>