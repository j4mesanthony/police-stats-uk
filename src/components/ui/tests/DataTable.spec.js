import { describe, it, expect } from 'vitest';
import DataTable from '../DataTable.vue';
import { createApp } from 'vue';
import { renderToString } from '@vue/server-renderer';

describe('DataTable', () => {
    it('should not regress (snapshot test)', async () => {
        const app = createApp(DataTable);
        const string = await renderToString(app);
        expect(string).toMatchSnapshot();
    })
})