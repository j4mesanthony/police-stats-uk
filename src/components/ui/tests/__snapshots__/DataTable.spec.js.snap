// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTable > should not regress (snapshot test) 1`] = `"<!--v-if-->"`;
